tests/test_question_templates/question_inputs/q16_workspaces/q16_workspaces.md

```diff
@@ -0,0 +1,142 @@
+
+        # Update the data object with a new dict
+        data.update(data2)
+    grade: |
+        files = { f['name']: f['contents'] for f in data['submitted_answers'].get('_files', []) }
+        data['feedback']['results'] = []
+        data['score'] = 0
+
+        with open('workspace/file1.txt', 'rb') as f:
+            file1_content = base64.b64encode(f.read()).decode()
+        with open('workspace/file2.txt', 'rb') as f:
+            file2_content = base64.b64encode(f.read()).decode()
+
+        if 'newdir/file1_new.txt' not in files:
+            data['feedback']['results'].append('newdir/file1_new.txt was not created.')
+        elif files['newdir/file1_new.txt'] != file1_content:
+            data['feedback']['results'].append('newdir/file1_new.txt was created, but it is not based on file1.txt.')
+        elif 'file1.txt' in files:
+            data['feedback']['results'].append('newdir/file1_new.txt was created properly, but the old file still exists.')
+            data['score'] += 0.2
+        else:
+            data['feedback']['results'].append('file1.txt moved successfully.')
+            data['score'] += 0.4
+
+        if 'newdir/file2_new.txt' not in files:
+            data['feedback']['results'].append('newdir/file2_new.txt was not created.')
+        elif files['newdir/file2_new.txt'] != file2_content:
+            data['feedback']['results'].append('newdir/file2_new.txt was created, but it is not based on file2.txt.')
+        elif 'file2.txt' not in files:
+            data['feedback']['results'].append('newdir/file2_new.txt was created properly, but the old file no longer exists.')
+            data['score'] += 0.2
+        else:
+            data['feedback']['results'].append('file2.txt copied successfully.')
+            data['score'] += 0.4
+
+        if files and 'filebad.txt' in files:
+            data['feedback']['results'].append('filebad.txt still exists.')
+        else:
+            data['feedback']['results'].append('filebad.txt deleted successfully.')
+            data['score'] += 0.2
+    prepare: |
+        pass
+    parse: |
+        if '_files' in data['format_errors']:
+            del data['format_errors']['_files']
+part1:
+  type: workspace
+---
+# {{ params.vars.title }}
+
+## Question Text
+
+Open the Workspace below and perform the following tasks.
+
+- Create a new directory called `newdir`;
+- Move the file `file1.txt` to the directory `newdir`, with the new name `file1_new.txt`;
+- Copy the file `file2.txt` to the directory `newdir`, with the new name `file2_new.txt`;
+- Delete the file `filebad.txt`.
+
+Once you complete these tasks, click on "Save & Grade" below.
+
+### Answer Section
+
+
+### pl-submission-panel
+
+<ul>
+  {{#feedback.results}}
+  <li>{{.}}</li>
+  {{/feedback.results}}
+</ul>
+
+<!-- Everything here will get inserted directly into the pl-submission-panel element at the end of the `question.html`.
+Please remove this section if it is not application for this question. -->
```

Comment:
Wanted to leave in the note here, but not have it show up, since this section has a valid input for this test